src/fragments/lib/graphqlapi/ios/working-with-files.mdx

```diff
@@ -0,0 +1,1893 @@
+            print("Failed with error: ", result)
+            return
+        }
+        guard var song = queriedSong else {
+            print("Song may have been deleted, no song by id: ", currentSong.id)
+            await setCurrentSong(nil)
+            return
+        }
+        guard let coverArtKey = song.coverArtKey else {
+            print("There is no cover art key to remove image association")
+            return
+        }
+        
+        // Set the association to nil and update it
+        song.coverArtKey = nil
+        let updateResult = try await Amplify.API.mutate(request: .update(song))
+        guard case .success(let updatedSong) = updateResult else {
+            print("Failed with error: ", result)
+            return
+        }
+        
+        // Remove the image
+        try await Amplify.Storage.remove(key: coverArtKey,
+                                         options: .init(accessLevel: .private))
+        
+        await setCurrentSong(updatedSong)
+        await setCurrentImage(nil)
+    }
+    
+    func deleteSongAndArtCover() async throws {
+        await setIsLoading(true)
+        defer {
+            Task {
+                await setIsLoading(false)
+            }
+        }
+        guard let currentSong = currentSong else {
+            print("There is no song to delete. Create a song first.")
+            return
+        }
+        
+        // Get the song record
+        let result = try await Amplify.API.mutate(request: .get(Song.self, byIdentifier: currentSong.id))
+        guard case .success(let queriedSong) = result else {
+            print("Failed with error: ", result)
+            return
+        }
+        guard let song = queriedSong else {
+            print("Song may have been deleted, no song by id: ", currentSong.id)
+            await setCurrentSong(nil)
+            return
+        }
+        
+        if let coverArt = song.coverArtKey {
+            // Remove the image
+            try await Amplify.Storage.remove(key: coverArt,
+                                             options: .init(accessLevel: .private))
+        }
+        
+        // Delete the song record
+        let deleteResult = try await Amplify.API.mutate(request: .delete(song))
+        guard case .success = deleteResult else {
+            print("Failed with error: ", deleteResult)
+            return
+        }
+        await setCurrentSong(nil)
+        await setCurrentImage(nil)
+    }
+    
+    @MainActor
+    func setCurrentSong(_ song: Song?) {
+        self.currentSong = song
+    }
+    
+    @MainActor
+    func setCurrentImage(_ image: UIImage?) {
+        self.currentImage = image
+    }
+
+    @MainActor
+    func setIsLoading(_ isLoading: Bool) {
+        self.isLoading = isLoading
+    }
+}
+
+struct SongView: View {
+    
+    @State private var isImagePickerPresented = false
+    @State private var songName: String = ""
+    
+    @StateObject var vm = SongViewModel()
```

Comment:
we can rename this to something verbose like `viewModel` or `songViewModel`

Reply:
i think `viewModel` is better, verbose enough but not necessarily need `song-` since there's only one in this context